### ISSUE TITLE
libstdcxx5, gcc throws a segmentation fault on compilation. 

### ISSUE THREAD

**Issue by pyrofolium** [NixOS/nixpkgs#170055]
### Describe the bug
libstdcxx5 throws segmentation fault on installation. 

### Steps To Reproduce
1. run `nix-env -iA nixpkgs.libstdcxx5 `
2. watch failure
3. ...

### Expected behavior
success

### Screenshots
no screenshots will post log

### Additional context
log:
```warning: name collision in input Nix expressions, skipping '/home/brian/.nix-defexpr/channels_root/unstable'
replacing old 'libstdc++5-3.3.6'
installing 'libstdc++5-3.3.6'
this derivation will be built:
  /nix/store/2q71g0wvsa00ik66kprqldv5iyc2lfdy-libstdc++5-3.3.6.drv
building '/nix/store/2q71g0wvsa00ik66kprqldv5iyc2lfdy-libstdc++5-3.3.6.drv'...
unpacking sources
unpacking source archive /nix/store/x8bl1fmzzhm7fkhhgk56m5pdb3sf4wby-gcc-core-3.3.6.tar.bz2
unpacking source archive /nix/store/5ykqhz8yflwfad51ayp0bjzp55kyajxs-gcc-g++-3.3.6.tar.bz2
source root is gcc-3.3.6
setting SOURCE_DATE_EPOCH to timestamp 1115126617 of file gcc-3.3.6/.brik
patching sources
applying patch /nix/store/3dagnp0z7nflgsdzgzgmjfw63j3k052q-no-sys-dirs.patch
patching file gcc/cppdefault.c
patching file gcc/gcc.c
patching file gcc/Makefile.in
applying patch /nix/store/jbilvhq21cp18za211ich1vjsy4xp0ki-siginfo.patch
patching file gcc/config/i386/linux.h
applying patch /nix/store/r7p5mwzspp8qaky2g6lyrsswdzps0yq7-gcc-3.4.3-no_multilib_amd64.patch
patching file gcc/config/i386/t-linux64
applying patch /nix/store/bimxmanbwfmp6ksj7nlv1a3dcwzcrzrb-struct-ucontext.patch
patching file gcc/config/alpha/linux.h
patching file gcc/config/i386/linux.h
patching file gcc/config/i386/linux64.h
patching file gcc/config/rs6000/linux.h
configuring
configure flags: --prefix=/nix/store/jf7wynjq7sdqxjvnv8h73b9i5aji23p3-libstdc++5-3.3.6 --disable-multilib --enable-__cxa-atexit --enable-threads=posix --enable-languages=c++ --enable-clocale=gnu
Configuring for a x86_64-unknown-linux-gnu host.
Created "Makefile" in /build/build using "mt-frag"
Configuring libiberty...
creating cache ../config.cache
checking whether to enable maintainer-specific portions of Makefiles... no
checking for makeinfo... no
checking for perl... no
checking host system type... x86_64-unknown-linux-gnu
checking build system type... x86_64-unknown-linux-gnu
checking for ar... ar
checking for ranlib... ranlib
checking for gcc... gcc
checking whether we are using GNU C... yes
checking whether gcc accepts -g... yes
checking whether gcc and cc understand -c and -o together... yes
checking for POSIXized ISC... no
checking for working const... yes
checking for inline... inline
checking for a BSD compatible install... /nix/store/hgl0ydlkgs6y6hx9h7k209shw3v7z77j-coreutils-9.0/bin/install -c
checking how to run the C preprocessor... gcc -E
checking for sys/file.h... yes
checking for sys/param.h... yes
checking for limits.h... yes
checking for stdlib.h... yes
checking for malloc.h... yes
checking for string.h... yes
checking for unistd.h... yes
checking for strings.h... yes
checking for sys/time.h... yes
checking for time.h... yes
checking for sys/resource.h... yes
checking for sys/stat.h... yes
checking for sys/mman.h... yes
checking for fcntl.h... yes
checking for alloca.h... yes
checking for sys/pstat.h... no
checking for sys/sysmp.h... no
checking for sys/sysinfo.h... yes
checking for machine/hal_sysinfo.h... no
checking for sys/table.h... no
checking for sys/sysctl.h... no
checking for sys/systemcfg.h... no
checking for sys/wait.h that is POSIX.1 compatible... yes
checking whether time.h and sys/time.h may both be included... yes
checking whether errno must be declared... no
checking for ANSI C header files... yes
checking for uintptr_t... no
checking for pid_t... yes
checking whether the C compiler (gcc -g -O2 ) works... yes
checking whether the C compiler (gcc -g -O2 ) is a cross-compiler... no
checking for asprintf... yes
checking for atexit... yes
checking for basename... yes
checking for bcmp... yes
checking for bcopy... yes
checking for bsearch... yes
checking for bzero... yes
checking for calloc... yes
checking for clock... yes
checking for ffs... yes
checking for getcwd... yes
checking for getpagesize... yes
checking for index... yes
checking for insque... yes
checking for memchr... yes
checking for memcmp... yes
checking for memcpy... yes
checking for memmove... yes
checking for memset... yes
checking for mkstemps... yes
checking for putenv... yes
checking for random... yes
checking for rename... yes
checking for rindex... yes
checking for setenv... yes
checking for sigsetmask... yes
checking for strcasecmp... yes
checking for strchr... yes
checking for strdup... yes
checking for strncasecmp... yes
checking for strrchr... yes
checking for strstr... yes
checking for strtod... yes
checking for strtol... yes
checking for strtoul... yes
checking for tmpnam... yes
checking for vasprintf... yes
checking for vfprintf... yes
checking for vprintf... yes
checking for vsprintf... yes
checking for waitpid... yes
checking whether alloca needs Cray hooks... no
checking stack direction for C alloca... -1
checking for vfork.h... no
checking for working vfork... yes
checking for _doprnt... no
checking for sys_errlist... no
checking for sys_nerr... no
checking for sys_siglist... no
checking for external symbol _system_configuration... no
checking whether canonicalize_file_name must be declared... yes
checking for getrusage... yes
checking for on_exit... yes
checking for psignal... yes
checking for strerror... yes
checking for strsignal... yes
checking for sysconf... yes
checking for times... yes
checking for sbrk... yes
checking for gettimeofday... yes
checking for realpath... yes
checking for canonicalize_file_name... yes
checking for pstat_getstatic... no
checking for pstat_getdynamic... no
checking for sysmp... no
checking for getsysinfo... no
checking for table... no
checking for sysctl... no
checking for unistd.h... (cached) yes
checking for getpagesize... (cached) yes
checking for working mmap... yes
checking for working strncmp... yes
updating cache ../config.cache
creating ./config.status
creating Makefile
creating testsuite/Makefile
creating config.h
Configuring gcc...
loading cache ../config.cache
checking LIBRARY_PATH variable... ok
checking GCC_EXEC_PREFIX variable... ok
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking build system type... x86_64-unknown-linux-gnu
checking for gcc... (cached) gcc
checking whether the C compiler (gcc -g -O2 ) works... yes
checking whether the C compiler (gcc -g -O2 ) is a cross-compiler... no
checking whether we are using GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking whether gcc and cc understand -c and -o together... (cached) yes
checking for gnatbind... no
checking for compiler driver that understands Ada... no
checking whether gcc accepts -Wno-long-long... yes
checking how to run the C preprocessor... (cached) gcc -E
checking for inline... (cached) inline
checking for volatile... yes
checking for long double... yes
checking for long long int... yes
checking for __int64... no
checking for built-in _Bool... yes
checking size of short... 2
checking size of int... 4
checking size of long... 8
checking size of long long... 8
checking execution character set... ASCII
checking whether make sets ${MAKE}... yes
checking whether a default assembler was specified... no
checking whether a default linker was specified... no
checking for GNU C library... yes
checking for mawk... no
checking for gawk... gawk
checking whether ln works... yes
checking whether ln -s works... yes
checking for ranlib... (cached) ranlib
checking for a BSD compatible install... (cached) /nix/store/hgl0ydlkgs6y6hx9h7k209shw3v7z77j-coreutils-9.0/bin/install -c
checking for ANSI C header files... (cached) yes
checking whether time.h and sys/time.h may both be included... (cached) yes
checking for working stdbool.h... yes
checking whether string.h and strings.h may both be included... yes
checking for sys/wait.h that is POSIX.1 compatible... (cached) yes
checking for limits.h... (cached) yes
checking for stddef.h... yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for time.h... (cached) yes
checking for fcntl.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/file.h... (cached) yes
checking for sys/time.h... (cached) yes
checking for sys/mman.h... (cached) yes
checking for sys/resource.h... (cached) yes
checking for sys/param.h... (cached) yes
checking for sys/times.h... yes
checking for sys/stat.h... (cached) yes
checking for direct.h... no
checking for malloc.h... (cached) yes
checking for langinfo.h... yes
checking for thread.h... no
checking for pthread.h... yes
checking for CHAR_BIT... yes
checking byte ordering... little-endian
checking floating point format... IEEE (little-endian)
checking for mktemp... yes
checking for makeinfo... no
configure: warning: 
*** Makeinfo is missing or too old.
*** Info documentation will not be built.
checking for recent Pod::Man... no
checking for flex... flex
checking for bison... bison
checking for collect2 libraries... none required
checking for library containing exc_resume... no
checking for library containing ldexp... none required
checking for preprocessor stringizing operator... yes
checking for inttypes.h... yes
checking for times... (cached) yes
checking for clock... (cached) yes
checking for dup2... yes
checking for kill... yes
checking for getrlimit... yes
checking for setrlimit... yes
checking for atoll... yes
checking for atoq... no
checking for sysconf... (cached) yes
checking for strsignal... (cached) yes
checking for putc_unlocked... yes
checking for fputc_unlocked... yes
checking for fputs_unlocked... yes
checking for fwrite_unlocked... yes
checking for fprintf_unlocked... no
checking for getrusage... (cached) yes
checking for nl_langinfo... yes
checking for lstat... yes
checking for scandir... yes
checking for alphasort... yes
checking for gettimeofday... (cached) yes
checking for mmap... yes
checking for ssize_t... yes
checking for uid_t in sys/types.h... yes
checking type of array argument to getgroups... gid_t
checking whether the printf functions support %p... yes
checking whether read-only mmap of a plain file works... yes
checking whether mmap from /dev/zero works... yes
checking for MAP_ANON(YMOUS)... yes
checking whether mmap with MAP_ANON(YMOUS) works... yes
checking for pid_t... (cached) yes
checking for vfork.h... (cached) no
checking for working vfork... (cached) yes
checking for iconv... yes
checking for iconv declaration... 
         extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
checking whether getenv is declared... yes
checking whether atol is declared... yes
checking whether sbrk is declared... yes
checking whether abort is declared... yes
checking whether atof is declared... yes
checking whether getcwd is declared... yes
checking whether getwd is declared... yes
checking whether strsignal is declared... yes
checking whether putc_unlocked is declared... yes
checking whether fputs_unlocked is declared... yes
checking whether fwrite_unlocked is declared... yes
checking whether fprintf_unlocked is declared... no
checking whether strstr is declared... yes
checking whether errno is declared... yes
checking whether vasprintf is declared... yes
checking whether malloc is declared... yes
checking whether realloc is declared... yes
checking whether calloc is declared... yes
checking whether free is declared... yes
checking whether basename is declared... yes
checking whether getopt is declared... yes
checking whether clock is declared... yes
checking whether getrlimit is declared... yes
checking whether setrlimit is declared... yes
checking whether getrusage is declared... yes
checking whether times is declared... yes
checking for struct tms... yes
checking for clock_t... yes
checking for uchar... 
checking for .preinit_array/.init_array/.fini_array support... yes
checking if mkdir takes one argument... no
checking for main in -lunwind... no
Using `../../gcc-3.3.6/gcc/config/i386/i386.c' for machine-specific logic.
Using `../../gcc-3.3.6/gcc/config/i386/i386.md' as machine description file.
Using the following target machine macro files:
        ../../gcc-3.3.6/gcc/config/i386/biarch64.h
        ../../gcc-3.3.6/gcc/config/i386/i386.h
        ../../gcc-3.3.6/gcc/config/i386/unix.h
        ../../gcc-3.3.6/gcc/config/i386/att.h
        ../../gcc-3.3.6/gcc/config/dbxelf.h
        ../../gcc-3.3.6/gcc/config/elfos.h
        ../../gcc-3.3.6/gcc/config/svr4.h
        ../../gcc-3.3.6/gcc/config/linux.h
        ../../gcc-3.3.6/gcc/config/i386/x86-64.h
        ../../gcc-3.3.6/gcc/config/i386/linux64.h
checking for library containing strerror... none required
checking for working const... (cached) yes
checking for off_t... yes
checking for size_t... yes
checking for working alloca.h... (cached) yes
checking for alloca... yes
checking whether we are using the GNU C Library 2.1 or newer... yes
checking for argz.h... yes
checking for limits.h... (cached) yes
checking for locale.h... yes
checking for nl_types.h... yes
checking for malloc.h... (cached) yes
checking for stddef.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/param.h... (cached) yes
checking for feof_unlocked... yes
checking for fgets_unlocked... yes
checking for getcwd... (cached) yes
checking for getegid... yes
checking for geteuid... yes
checking for getgid... yes
checking for getuid... yes
checking for mempcpy... yes
checking for munmap... yes
checking for putenv... (cached) yes
checking for setenv... (cached) yes
checking for setlocale... yes
checking for stpcpy... yes
checking for strchr... (cached) yes
checking for strcasecmp... (cached) yes
checking for strdup... (cached) yes
checking for strtoul... (cached) yes
checking for tsearch... yes
checking for __argz_count... yes
checking for __argz_stringify... yes
checking for __argz_next... yes
checking for iconv... (cached) yes
checking for iconv declaration... (cached) 
         extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
checking for nl_langinfo and CODESET... yes
checking for LC_MESSAGES... yes
checking whether NLS is requested... yes
checking whether included gettext is requested... no
checking for libintl.h... yes
checking for GNU gettext in libc... yes
checking for dcgettext... yes
checking for msgfmt... no
checking for gmsgfmt... :
checking for xgettext... no
checking for bison... bison
checking version of bison... 3.8.2, ok
checking for catalogs to be installed...  be ca da de el es fr ja nl rw sv tr be ca da de el es fr ja nl rw sv tr
checking what assembler to use... as
checking what linker to use... ld
checking what nm to use... nm
checking what objdump to use... objdump
checking assembler alignment features... .p2align including maximum skip
checking assembler subsection support... working .subsection -1
checking assembler weak support... yes
checking assembler hidden support... yes
checking assembler leb128 support... ../../gcc-3.3.6/gcc/configure: line 7351: test: too many arguments
yes
checking assembler eh_frame optimization... yes
checking assembler section merging support... yes
checking assembler thread-local storage support... yes
checking assembler instructions... filds fists 
checking cmov syntax... no
checking assembler GOTOFF in data directives... no
checking assembler dwarf2 debug_line support... yes
checking assembler --gdwarf2 support... yes
checking assembler --gstabs support... yes
checking linker read-only and read-write section mixing... read-write
checking linker PT_GNU_EH_FRAME support... yes
checking linker --as-needed support... yes
Using ggc-page for garbage collection.
checking whether to enable maintainer-specific portions of Makefiles... no
Links are now set up to build a native compiler for x86_64-unknown-linux-gnu
updating cache ../config.cache
creating ./config.status
creating Makefile
creating intl/Makefile
creating fixinc/Makefile
creating gccbug
creating mklibgcc
creating mkheaders
creating auto-host.h
building
cat: /nix/store/dgic5ks4yixhh0havidjwd02rskmqlgp-binutils-wrapper-2.38/nix-support/libc-ldflags-before: No such file or directory
build flags: SHELL=/nix/store/fcd0m68c331j7nkdxvnnpb8ggwsaiqac-bash-5.1-p16/bin/bash NATIVE_SYSTEM_HEADER_DIR=/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include SYSTEM_HEADER_DIR=/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include CFLAGS_FOR_BUILD=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib CFLAGS_FOR_TARGET=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib CXXFLAGS_FOR_BUILD=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib CXXFLAGS_FOR_TARGET=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib FLAGS_FOR_TARGET=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib LDFLAGS_FOR_BUILD=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib LDFLAGS_FOR_TARGET=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib BOOT_CFLAGS=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib BOOT_LDFLAGS=-I/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -B/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/\ \ -idirafter\ /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include\ -idirafter\ /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/lib/gcc/x86_64-unknown-linux-gnu/11.2.0/include-fixed\ \ -O2\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-rpath\ -Wl\,/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib\ -Wl\,-L/nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib
make[1]: Entering directory '/build/build/libiberty'
if [ x"" != x ] && [ ! -d pic ]; then \
  mkdir pic; \
else true; fi
touch stamp-picdir
if [ x"" != x ]; then \
  gcc -c -DHAVE_CONFIG_H -g -O2 -I. -I../../gcc-3.3.6/libiberty/../include  -W -Wall -Wtraditional -pedantic  ../../gcc-3.3.6/libiberty/regex.c -o pic/regex.o; \
else true; fi
gcc -c -DHAVE_CONFIG_H -g -O2 -I. -I../../gcc-3.3.6/libiberty/../include  -W -Wall -Wtraditional -pedantic ../../gcc-3.3.6/libiberty/regex.c -o regex.o
In file included from /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include/features.h:488,
                 from /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include/stdlib.h:25,
                 from ../../gcc-3.3.6/libiberty/regex.c:130:
/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include/sys/cdefs.h:1: internal compiler error: Segmentation fault
    1 | /* Copyright (C) 1992-2021 Free Software Foundation, Inc.
      | 
0x15f6767 internal_error(char const*, ...)
        ???:0
0x16238d4 cpp_sys_macro_p(cpp_reader*)
        ???:0
0x161391d cpp_classify_number(cpp_reader*, cpp_token const*, char const**, unsigned int)
        ???:0
0x161554a _cpp_parse_expr
        ???:0
0x160fe88 _cpp_handle_directive
        ???:0
0x161ea48 _cpp_lex_token
        ???:0
0x746a0e c_lex_with_flags(tree_node**, unsigned int*, unsigned char*, int)
        ???:0
0x70087c c_parse_file()
        ???:0
0x74f21d c_common_parse_file()
        ???:0
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [Makefile:126: regex.o] Error 1
make[1]: Leaving directory '/build/build/libiberty'
make: *** [Makefile:1293: all-libiberty] Error 2
error: builder for '/nix/store/2q71g0wvsa00ik66kprqldv5iyc2lfdy-libstdc++5-3.3.6.drv' failed with exit code 2;
       last 10 log lines:
       >       ???:0
       > 0x74f21d c_common_parse_file()
       >  ???:0
       > Please submit a full bug report,
       > with preprocessed source if appropriate.
       > Please include the complete backtrace with any bug report.
       > See <https://gcc.gnu.org/bugs/> for instructions.
       > make[1]: *** [Makefile:126: regex.o] Error 1
       > make[1]: Leaving directory '/build/build/libiberty'
       > make: *** [Makefile:1293: all-libiberty] Error 2
       For full logs, run 'nix log /nix/store/2q71g0wvsa00ik66kprqldv5iyc2lfdy-libstdc++5-3.3.6.drv'.
```

**Comment by crimsonalucard** [comment]
@abbradar

**Comment by crimsonalucard** [comment]
It looks like gcc is crashing. Perhaps updating the version of gcc will fix?

**Comment by crimsonalucard** [comment]
The offending line:

`/nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include/sys/cdefs.h:1: internal compiler error: Segmentation fault`

**Comment by trofi** [comment]
That looks like a `gcc-11.2.0` bug in preprocessor (possibly related to `-Wtraditional` handling):

```
$ echo '#include <strings.h>' | LANG=C gcc -E -Wtraditional -fsyntax-only - >/dev/null

In file included from /nix/store/l1cds4dpaad47f1n6jwzva2sn1dzdn6a-glibc-2.34-115-dev/include/features.h:488,
                 from /nix/store/l1cds4dpaad47f1n6jwzva2sn1dzdn6a-glibc-2.34-115-dev/include/strings.h:21,
                 from <stdin>:1:
/nix/store/l1cds4dpaad47f1n6jwzva2sn1dzdn6a-glibc-2.34-115-dev/include/sys/cdefs.h:1: internal compiler error: Segmentation fault
    1 | /* Copyright (C) 1992-2021 Free Software Foundation, Inc.
      |
```

I'll try to extract something self-contained and check if newer `gcc` versions are affected.

**Comment by trofi** [comment]
Minimal example (`cdefs.h`):

```c
#define __glibc_has_attribute(attr) __has_attribute(attr)
#if __glibc_has_attribute(__deprecated__)
#endif
```

Crash:

```
LANG=C gcc -E cdefs.h -Wtraditional -nostdinc
# 0 "cdefs.h"
# 0 "<built-in>"
# 0 "<command-line>"
# 1 "cdefs.h"
cdefs.h:1: internal compiler error: Segmentation fault
    1 | #define __glibc_has_attribute(attr) __has_attribute(attr)
      |
0x15a1667 internal_error(char const*, ...)
        ???:0
0x15ce7d4 cpp_sys_macro_p(cpp_reader*)
        ???:0
```

**Comment by trofi** [comment]
Upstream `gcc-11.3.0` works (released recently), `gcc-11.2.0` crashes. Bisect gave https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=408d88af60e3268f7fad59fa393ec7e28922c435 which is https://gcc.gnu.org/PR101638.

I think it's reasonable to force `gcc-10`'s stdenv for `libstdcxx5` until we get fixed `gcc-11` into `nixpkgs`.

**Comment by trofi** [comment]
Let's try https://github.com/NixOS/nixpkgs/pull/170117 as a workaround.

**Comment by trofi** [comment]
Longer term `gcc-11.3.0` update: https://github.com/NixOS/nixpkgs/pull/170121

**Comment by lf-** [comment]
I believe this is fixed since `libstdcxx5` builds correctly on 8a3d063a494c4b8c767190a5ce3e4075a75f9d07
